playground/backend/internal/code_processing/code_processing.go

```diff
@@ -399,13 +399,13 @@ func cancelCheck(ctx context.Context, pipelineId uuid.UUID, cancelChannel chan b
 			return
 		case <-ticker.C:
 			cancel, err := cacheService.GetValue(ctx, pipelineId, cache.Canceled)
-			if err != nil {
-				logger.Errorf("%s: Error during getting value from the cache: %s", pipelineId, err.Error())
-			}
-			if cancel.(bool) {
+			if err == context.DeadlineExceeded || (err == nil && cancel.(bool)) {
 				cancelChannel <- true
 				return
 			}
```

Comment:
Is there a reason we are not following this pattern: https://go.dev/play/p/j9Edbjx2y5x?